lsp-yaml.el

```diff
@@ -135,5 +153,68 @@
                                       (lsp--set-configuration
                                        (lsp-configuration-section "yaml"))))))
 
+(defconst lsp-yaml--built-in-kubernetes-schema
+  '((name . "Kubernetes")
+    (description . "Built-in kubernetes manifest schema definition")
+    (url . :kubernetes)
+    (fileMatch . ["*-k8s.yaml" "*-k8s.yml"])))
+
+(defun lsp-yaml-download-schema-store-db (&optional force-downloading)
+  "Download the remote schema store at `lsp-yaml-schema-store-uri' into local cache.
+Set FORCE-DOWNLOADING to non-nil to force re-download the database."
+  (interactive "P")
+  (or force-downloading
+      (file-exists-p lsp-yaml-schema-store-local-db)
+      (url-copy-file lsp-yaml-schema-store-uri lsp-yaml-schema-store-local-db)))
+
+(defun lsp-yaml--get-supported-schemas ()
+  "Get out the list of supported schemas."
+  (progn
+    (if lsp-yaml-schema-store-enable
+        (if (not lsp-yaml--schema-store-schemas-alist)
+            (progn
+              (lsp-yaml-download-schema-store-db)
+              (setq lsp-yaml--schema-store-schemas-alist
+                    (alist-get 'schemas (json-read-file lsp-yaml-schema-store-local-db))))))
+    (seq-concatenate 'list (list lsp-yaml--built-in-kubernetes-schema) lsp-yaml--schema-store-schemas-alist)))
+
+(defun lsp-yaml--helm-schema-candidates ()
+  "Build helm source for schema candidates."
+  (cl-loop for schema in (lsp-yaml--get-supported-schemas)
+	   collect (cons
+		    (alist-get 'description schema)
+		    schema)))
+
+(defun lsp-yaml-set-buffer-schema (uri)
+  "Set yaml schema for the current buffer to URI."
+  (let* ((workspace-path (file-relative-name
+                          (lsp--uri-to-path (lsp--buffer-uri))
+                          (lsp-workspace-root (lsp--buffer-uri))))
+         (glob-pattern (concat "/" workspace-path))
+         (current-config (cl-assoc uri lsp-yaml-schemas :test 'equal))
+         (current-patterns (and current-config (cdr current-config))))
+    (if current-config
+        (or (cl-member glob-pattern current-patterns :test 'equal)
+            (setq lsp-yaml-schemas
+                  (cl-acons uri (cl-list* glob-pattern current-patterns)
+                            (cl-remove current-config lsp-yaml-schemas))))
+      (setq lsp-yaml-schemas
+            (cl-acons uri (cl-list* glob-pattern nil) lsp-yaml-schemas)))
+    (lsp-register-custom-settings
```

Comment:
this is already registered.

Reply:
Since we modify the 'yaml.schemas' to adding a new file. I want to updating that specific path of the current setting.

Reply:
Not sure if there is any better way to do it.
Would you mind to give me a hint.

Reply:
What I am saying is that you can remove lsp-register-custom-settings call and it should work.

Reply:
Yes, you're right.